collator/src/collator/mod.rs

```diff
@@ -916,46 +933,61 @@ impl CollatorStdImpl {
             .get_last_imported_anchor_id_and_ct()
             .unwrap_or_default();
 
+        // do not import anchor if mempool may be paused
+        // needs to process more anchors in collator first
+        if id - top_processed_to_anchor > max_consensus_lag_rounds / 2 {
```

Comment:
no possibility for overflow here?  
for example, if we collated, then synced, then began collating again, and no new anchors were imported from mempool (local mempool disconnected or is syncing) 
maybe `saturating_sub`?